### DISCUSSION TITLE
Byedpi OpenWrt Luci-app

### DISCUSSION THREAD

**Discussion by Yusupoff** [hufrea/byedpi#232] [Ideas]
Помогите разобраться со скриптами lua. Хочу сделать панель для byedpi в openwrt есть пример но она не видит файл конфигурации `/etc/config/byedpi`

"/usr/lib/lua/luci/model/cbi/byedpi.lua"
```lua
local sys = require "luci.sys"
local m = Map("byedpi", "byedpi", "Описание")
local s = m:section(NamedSection, "byedpi", "byedpi", "byedpi", "Описание")

local o
s:option(Flag, "enabled", "enabled", "Текст")
o = s:option(Value, "cmd_opts", "cmd opts", "Описание")
 
local bs = m:section(NamedSection, "byedpi", "byedpi", "Service status")

local asts = sys.call("/etc/init.d/byedpi enabled &>/dev/null")

if asts == 0 then
	local asto = bs:option(Button, "_autostart_disable", "Autostart")
	asto.inputstyle = "negative"
	asto.inputtitle = "Disable"

	asto.write = function(self, section)
		sys.call("/etc/init.d/byedpi disable &>/dev/null")
	end
else
	local asto = bs:option(Button, "_autostart_enable", "Autostart")
	asto.inputstyle = "positive"
	asto.inputtitle = "Enable"

	asto.write = function(self, section)
		sys.call("/etc/init.d/byedpi enable &>/dev/null")
	end
end

local sts = sys.call("/etc/init.d/byedpi running &>/dev/null")

if sts == 0 then
	local sto = bs:option(Button, "_status_stop", "Status")
	sto.inputstyle = "negative"
	sto.inputtitle = "Stop"
	sto.description = "byedpi is currently active"

	sto.write = function(self, section)
		sys.call("/etc/init.d/byedpi stop &>/dev/null")
	end
else
	local sto = bs:option(Button, "_status_start", "Status")
	sto.inputstyle = "positive"
	sto.inputtitle = "Start"
	sto.description = "byedpi is currently down"

	sto.write = function(self, section)
		sys.call("/etc/init.d/byedpi start &>/dev/null")
	end
end

local rso = bs:option(Button, "_restart", "Restart")
rso.inputstyle = "action"
function rso.write(self, section)
	sys.call("/etc/init.d/byedpi restart &>/dev/null")
end

local fwo = bs:option(Button, "_firewall", "Firewall")
fwo.inputtitle = "Reload"
fwo.inputstyle = "action"
function fwo.write(self, section)
	sys.call("/etc/init.d/firewall reload")
end

local logs = sys.exec("logread -l 800 -p ciadpi | grep ciadpi | sed '1!G;h;$!d'")
local o = bs:option(DummyValue, "_logs", "Logs")
o.rawhtml = true
o.value = logs
o.wrap = "off"
o.rows = 33
o.readonly = true
o.template = "cbi/tvalue"
o.width = "100%"

return m
``` 

"/usr/lib/lua/luci/controller/byedpi.lua"
```lua
module("luci.controller.byedpi", package.seeall)

function index()
	entry( {"admin", "services", "byedpi"}, cbi("byedpi"), _("main"))
end
```

**Comment by Yusupoff** [comment]
`/etc/config/byedpi`
```
config byedpi 'main'
	option enabled 1

#	if set, all other uci settings are ignored
	option cmd_opts '--split 1 --disorder 3+s --mod-http=h,d --auto=torst --tlsrec 1+s'
```

**Reply by Yusupoff** [reply]
https://github.com/spvkgn/ByeDPI-OpenWrt

**Reply by The-city-not-present** [reply]
Добрый день

Вы запостили ссылку, но совсем не понятно, что это. Там тоже описание отсутствует. Ну можно же было хоть короткую строчку оставить, что это. Окей, какой-то пакет, который я могу установить. Он с gui? Он запускается из командной строки? В папке "/usr/lib/lua/luci/controller/" никакого файла с "bye" в имени файла нет. Окей, это не дополнение к luci (наверно). Я ведь не знаю, какие файлы для неё нужны. Но и вообще никакого исполнямого файла byedpi в системе не появилось. Если набрать `find -type f -iname "byedpi* /`, вроде исполняемого файла в системе не вижу. Есть `/etc/init.d/byedpi`. Вот вы постите ссылку без описания, другой публикует репозиторий без описания - разве сложно было добавить короткую строку, что это?

**Comment by The-city-not-present** [comment]
Хммм, подскажете, что дальше

Набрал
```
root@OpenWrt:~/downloads# opkg install byedpi_0.16.6-r1_mipsel_24kc.ipk
Installing byedpi (0.16.6-r1) to root...
Configuring byedpi.
root@OpenWrt:~/downloads#
```

Перезагрузил роутер.

Должно что-то появиться в меню? Кликаю, кликаю, не вижу. Файл `/etc/config/byedpi` существует, но идея же в том, чтобы в Luci появилось?

**Reply by bam80** [reply]
Такое ощущение что правила настройки файерволла в пакете отсутствуют.
Есть идеи, может это где-то описано?
@spvkgn

**Reply by spvkgn** [reply]
> Такое ощущение что правила настройки файерволла в пакете отсутствуют.

Настройки файрволла для чего? И так должно работать - https://github.com/hufrea/byedpi/tree/main/dist/linux

**Reply by bam80** [reply]
Я не учел что в отличие от других решений, byedpi это прокся. Поэтому да, видимо файрволл здесь ни при чем.
